### DISCUSSION TITLE
[Airflow 1.10.15] Airflow looks for a variable that doesn't exist

### DISCUSSION THREAD

**Discussion by lozuwa** [apache/airflow#22716] [Q&A]
I was debugging a DAG and renamed one of its variables. After that I put the variable name back to its previous name. However, airflow keeps complaining about my DAG pointing to the previous variable name (see the image).

![Screen Shot 2022-04-03 at 18 56 31](https://user-images.githubusercontent.com/11334414/161452494-127d093d-c4da-4739-8bab-276312b9e74b.png)

Things I have done but the pop up is still there,
- I am using git-sync for DAGs. I validated the change was pushed to the container.
- I've configured dag_dir_list_interval to 60 seconds. It's been 3 hours and the same message pops up. 
- I have restarted the webserver, workers and scheduler pods. 

Does anyone know how this part works? Maybe I can fix it by deleting some reference in the database or something.

**Comment by potiuk** [comment]
It should get cleaned when the dag is parsed again - but maybe you have different dag id in your dag now, or the file still remains somewhere in  your DAG folder in scheduler. It could also be (though that's quite unlikely) generated in .pyc file that could be left-over from some previous run or manual compiling of the python code (remove the .pyc files then). 

If this is the last case and - for some reason - your pyc files got "stale" (not sure why it could happen - but it could be if you have not synchronized time on different servers of yours for example) then setting [PYTHONDONTWRITEBYTECODE](https://docs.python.org/3/using/cmdline.html#envvar-PYTHONDONTWRITEBYTECODE) variable might prevent from those .pyc files ot be generated.

**Reply by lozuwa** [reply]
Hey, thanks for answering.

1. **.py files** .- I am working on airflow kubernetes. So persistences are volatile. Restarting the pod makes all files to go away. Also, I am using git-sync so the latest commit on my dag repository is always pulled. 
2. The dag parsing is set to 60 seconds. It's been a day and the annoying message is still there.
3. I think this is db related. Because it's the only thing that is stateful in the whole deployment.

**Reply by potiuk** [reply]
First of all. I heartily recommend to use our official  Hem Chart and move to Airflow 2. Airflow 1.10 is End Of Life as of June 2021 and there are a number of security fixes only released in Airflow 2. 

Running Airflow 1.10 you make yourself and your company vulnerable.

By doing your own deployment, you are also trying to solve problems that have already been solved in our Helm Chart: https://airflow.apache.org/docs/helm-chart/stable/index.html

In this case the problem is that you should never run scheduler before the first git-syns happens if you have no persistency. This is already solved in the Helm Chart of ours via init containers.

**Reply by potiuk** [reply]
Generally speaking - migrate.https://airflow.apache.org/docs/apache-airflow/stable/upgrading-from-1-10/index.html

**Reply by lozuwa** [reply]
I am using the official helm chart with an extended docker image. 

I figured out the problem, I changed the fernet key during the upgrade to 1.10.15. That's why I had issues with all key/values that had encryption.

**Reply by potiuk** [reply]
Good to know. Still my advice to migrate to Airflow 2 is more than valid.

**Reply by lozuwa** [reply]
I know! Soon I hope.
